src/NHibernate/Cfg/Configuration.cs

```diff
@@ -221,7 +221,7 @@ public string GetIdentifierPropertyName(string className)
 			public IType GetReferencedPropertyType(string className, string propertyName)
 			{
 				PersistentClass pc = GetPersistentClass(className);
-				Property prop = pc.GetProperty(propertyName);
+				Property prop = pc.GetReferencedProperty(propertyName);
```

Comment:
`GetProperty` yields properties of the entity, it does not go inside components for yielding the component property.